### PR TITLE
feat: Action bar refactor to match fiori 3

### PR DESCRIPTION
**Pull request by manu-kr**
## Related Issue

https://github.com/SAP/fundamental-styles/issues/565
## Description

added properties and values to match Fiori 3.
documentation updated.
test scenarios simplified.

### Before:
image

### After:
image

after npm run test:update
image

### CODE REVIEW

**Comment by droshev** [comment]
@manu-kr I see that you added 2 images for the nested list. Is this intentional?

**Comment by manu-kr** [comment]
> @manu-kr I see that you added 2 images for the nested list. Is this intentional?

>@droshev I will check on this.

File: src/action-bar.scss [hunk 1]
```diff
@@ -72,42 +91,46 @@ $block: #{$fd-namespace}-action-bar;
...
-    margin-bottom: fd-space(base);
+    height: 2rem;
+    padding-right: 0.75rem;
+    vertical-align: middle;
+    margin: 0;
   }
 
   &__description {
     @include fd-reset();
     @include fd-var-color("color", $fd-action-bar-description-color, --fd-color-text-3);
```
[the remark this answers is not in the document]
Comment (manu-kr) [reply src/action-bar.scss]: Sure i will remove this.

**Comment by manu-kr** [comment]
> > @manu-kr I see that you added 2 images for the nested list. Is this intentional?
> 
> > @droshev I will check on this.

These files are committed unintentionally and changes are reverted and committed back previous version'

File: src/action-bar.scss [hunk 2]
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
...
   @include fd-reset();
 
   background-color: $fd-action-bar-background-color;
   min-height: $fd-action-bar-height;
-  padding-right: 0;
-  padding-left: 0;
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
   display: flex;
+  display: -webkit-flex;
```
Comment (jbadan) [inline src/action-bar.scss]: PostCSS should be handling this for you - we shouldn't need to add webkit fallbacks manually.

File: src/action-bar.scss [hunk 3]
```diff
@@ -72,42 +92,49 @@ $block: #{$fd-namespace}-action-bar;
...
   &__title {
     @include fd-reset();
     @include fd-heading();
-    @include fd-type("4");
-    @include fd-action-bar-responsive();
 
-    margin-bottom: fd-space(base);
+    font-size: var(--sapFontHeader3Size);
+    font-family: var(--sapFontHeaderFamily);
+    font-weight: normal;
```
Comment (jbadan) [inline src/action-bar.scss]: Shouldn't the font-weight be in the reset?

Comment (jbadan) [reply src/action-bar.scss]: See https://github.com/SAP/fundamental-styles/blob/master/src/mixins/_mixins.scss#L10

File: src/action-bar.scss [hunk 4]
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
     }
   }
 
+  @include fd-screen(s) {
+    padding-left: map_get($fd-action-bar-padding-x, "s");
```
Comment (stefanoScalzo) [inline src/action-bar.scss]: where are you getting these measurements

Comment (manu-kr) [reply src/action-bar.scss]: All these measurements based on the fiori 3 standards and here is the new design prepared based on the fiori 3 specs like typography spacings etc.
https://sapariba.invisionapp.com/share/7WSNCMKJQHP#/screens/370170777_ActionBar

File: src/action-bar.scss (hunk 2 above)
Comment (manu-kr) [reply src/action-bar.scss]: removed this.

File: src/action-bar.scss (hunk 3 above)
Comment (manu-kr) [reply src/action-bar.scss]: Yes you're right. Fixed this by removing font-weight:normal;

**Comment by manu-kr** [comment]
> please make sure that all discussions are addressed

Sure, I have addressed  all the issue.

**Comment by droshev** [comment]
@manu-kr The PR looks good to me. Can you add examples on the documentation page with cozy buttons?

**Comment by InnaAtanasova** [comment]
In the visuals that you provided a link to, the description is shown to go till the end of the container (under the buttons). 
Screen Shot 2020-02-18 at 3 06 06 PM

In the implementation the description container has the same width as the header container and after that truncates. 
Screen Shot 2020-02-18 at 2 54 45 PM

I didn't see any rules about the truncation of the description. Are you sure it needs to be truncated? Either way the "container" should go bellow the actions and then eventually truncate.

**Comment by InnaAtanasova** [comment]
There's problem with the title alignment:
Screen Shot 2020-02-18 at 3 04 47 PM
Screen Shot 2020-02-18 at 3 04 52 PM

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-02-18 at 3 17 11 PM

The mobile example should have 1rem padding left and right

**Comment by InnaAtanasova** [comment]
I think someone from your team mentioned that you need the title to be editable??? Do you still need this functionality?

Screen Shot 2020-02-18 at 3 27 39 PM

**Comment by InnaAtanasova** [comment]
Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?

Screen Shot 2020-02-18 at 3 30 39 PM

File: src/action-bar.scss [hunk 5]
```diff
@@ -43,25 +41,47 @@ $block: #{$fd-namespace}-action-bar;
...
+  flex-direction: row;
   align-items: baseline;
+  word-wrap: break-word;
+  justify-content: space-between;
+  padding-left: map_get($fd-action-bar-padding-x, "m");
+  padding-right: map_get($fd-action-bar-padding-x, "m");
 
   &__header {
+    flex-grow: 1;
+    width: 75%;
```
Comment (InnaAtanasova) [inline src/action-bar.scss]: this code should be after the reset function

**Comment by InnaAtanasova** [comment]
Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

**Comment by manu-kr** [comment]
> In the visuals that you provided a link to, the description is shown to go till the end of the container (under the buttons).
> Screen Shot 2020-02-18 at 3 06 06 PM
> 
> In the implementation the description container has the same width as the header container and after that truncates.
> Screen Shot 2020-02-18 at 2 54 45 PM
> 
> I didn't see any rules about the truncation of the description. Are you sure it needs to be truncated? Either way the "container" should go bellow the actions and then eventually truncate.

@InnaAtanasova There is requirement to truncate, description as well.I have modified the structure as per the visual spec

**Comment by manu-kr** [comment]
> I think someone from your team mentioned that you need the title to be editable??? Do you still need this functionality?
> 
> Screen Shot 2020-02-18 at 3 27 39 PM

@InnaAtanasova 
Yes , there is a requirement we are going to consider this in upcoming sprint.

File: src/action-bar.scss (hunk 5 above)
Comment (manu-kr) [reply src/action-bar.scss]: Placed after reset

**Comment by manu-kr** [comment]
> Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?
> 
> Screen Shot 2020-02-18 at 3 30 39 PM

Added example with cozy button for mobile example

**Comment by manu-kr** [comment]
> There's problem with the title alignment. The text is not centered vertically.
> Screen Shot 2020-02-18 at 3 04 47 PM
> Screen Shot 2020-02-18 at 3 04 52 PM

@InnaAtanasova ,

FIxed. and now its aligned properly. 

> There's problem with the title alignment. The text is not centered vertically.
> Screen Shot 2020-02-18 at 3 04 47 PM
> Screen Shot 2020-02-18 at 3 04 52 PM

@InnaAtanasova 
Aligned .

**Comment by manu-kr** [comment]
> Not sure the dimensions should stay the same for Mobile. Right now you are using buttons in compact mode and not the cosy mode. For a11y requirements the buttons in mobile should be cosy, which will prob affect the height of the action bar. Could you please check this with the designers?
> 
> Screen Shot 2020-02-18 at 3 30 39 PM

Sure i will check

**Comment by stefanoScalzo** [comment]
> Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

You removed the media queries but I don't seem to see the class modifiers?

**Comment by manu-kr** [comment]
> > Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html
> 
> You removed the media queries but I don't seem to see the class modifiers?

@stefanoScalzo 

Now i have removed the media queries. and added modifier classes for responsive behaviour.

**Comment by manu-kr** [comment]
> Instead of using media queries you can add modifier classes for the responsive behavior and let the user add these classes based on their needs. You can check the Page responsive behavior https://sap.github.io/fundamental-styles/layouts/page-layout.html

@InnaAtanasova 

I have removed media queries and added modifier classes for the responsive behaviour.

**Comment by manu-kr** [comment]
> There's problem with the title alignment. The text is not centered vertically.
> Screen Shot 2020-02-18 at 3 04 47 PM
> Screen Shot 2020-02-18 at 3 04 52 PM

Fixed.

**Comment by droshev** [comment]
@manu-kr can you fix the pipeline?

**Comment by manu-kr** [comment]
> @manu-kr can you fix the pipeline?

 Fixed the pipeline.

**Comment by droshev** [comment]
@manu-kr your PR is touching 120 files, which is not normal. Also the PR is updating the images for components that you didn't touch.
Would you update the PR?

**Comment by manu-kr** [comment]
This PR changes are moved to the PR #701 

https://github.com/SAP/fundamental-styles/pull/701

and closing this PR

**Pull request closed without merging**